temporal-sdk/src/main/java/io/temporal/client/WorkflowStubImpl.java

```diff
@@ -307,65 +305,108 @@ public <R> R update(
       Type resultType,
       Object... args) {
     checkStarted();
-    UpdateOutput<R> result;
-    WorkflowExecution targetExecution = execution.get();
     try {
-      result =
-          workflowClientInvoker.update(
-              new WorkflowClientCallsInterceptor.UpdateInput<>(
-                  targetExecution,
-                  updateName,
-                  updateId,
-                  args,
-                  resultClass,
-                  resultType,
-                  firstExecutionRunId));
-      return result.getResult();
-
-    } catch (Exception e) {
-      Throwable throwable = throwAsWorkflowFailureException(e, targetExecution);
-      throw new WorkflowServiceException(targetExecution, workflowType.orElse(null), throwable);
+      return startUpdate(
+              updateName,
+              updateId,
+              firstExecutionRunId,
+              UpdateWorkflowExecutionLifecycleStage
+                  .UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_COMPLETED,
+              resultClass,
+              resultType,
+              args)
+          .getResultAsync()
+          .get();
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
+    } catch (ExecutionException e) {
+      throw new RuntimeException(e);
     }
   }
 
   @Override
   public <R> UpdateHandle<R> startUpdate(String updateName, Class<R> resultClass, Object... args) {
-    return startUpdate(updateName, "", "", resultClass, resultClass, args);
+    return startUpdate(
+        updateName,
+        "",
+        "",
+        UpdateWorkflowExecutionLifecycleStage.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ACCEPTED,
+        resultClass,
+        resultClass,
+        args);
   }
 
   @Override
   public <R> UpdateHandle<R> startUpdate(
       String updateName,
       String updateId,
       String firstExecutionRunId,
+      UpdateWorkflowExecutionLifecycleStage waitPolicy,
       Class<R> resultClass,
       Type resultType,
       Object... args) {
     checkStarted();
     WorkflowExecution targetExecution = execution.get();
+    try {
+      if (updateId == null || updateId.isEmpty()) {
+        updateId = UUID.randomUUID().toString();
```

Comment:
@mmcshane We should probably do this in Go as well, instead of letting the server generate an ID.

Reply:
Hrmm, this is worth a discussion. What is the benefit of generating client side if the server side must return a result anyways before you can return a handle?

Reply:
Generating server side can be dangerous if the client retries

Reply:
Ok, I completely agree. I know other APIs have `request_id`, but IIRC we are intentionally choosing to use update ID as that ID here. Yes, server side we should probably require this and client side we should default it.